Plugin/DatatransPlugin.php

```diff
@@ -87,12 +88,22 @@ public function approve(FinancialTransactionInterface $transaction, $retry)
         $payConfirmParameter = new PayConfirmParameter();
 
         $request = $this->requestStack->getMasterRequest();
-        if ($request !== null && $request->request->has('responseCode') ) {
+        if ($request !== null
+            && $request->request->has(PayConfirmParameter::PAY_PARAM_RESPONSECODE)) {
+
+            if ($request->request->get(PayConfirmParameter::PAY_PARAM_RESPONSECODE)
+                != PayConfirmParameter::PAY_PARAM_RESPONSECODE_SUCCESS) {
+
+                $payConfirmParameter = $this->client->getConfirmParameter($request->request);
```

Comment:
Why call `getConfirmParameter()` and assign the result to `$payConfirmParameter` if en exception gets thrown by the next line?